### ISSUE TITLE
Story only renders on hot reloading

### ISSUE THREAD

**Issue by im-me** [storybookjs/storybook#18859]
**Describe the bug**
Some stories only render on hot reloading.

On first load and on page reloading (cmd + r) the affected stories disappear from the list.
Only after some changes in the files, the stories reappear after hot reloading.

**To Reproduce**
Difficult to reproduce, since the bug occures only for one component.

**System**
 System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
  Binaries:
    Node: 16.3.0 - ~/.nvm/versions/node/v16.3.0/bin/node
    Yarn: 1.22.11 - ~/.nvm/versions/node/v16.3.0/bin/yarn
    npm: 8.5.4 - ~/.nvm/versions/node/v16.3.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.134

**Additional context**

**Comment by shilman** [comment]
We’re cleaning house! Storybook has changed a lot since this issue was created and we don’t know if it’s still valid. Please open a new issue referencing this one if:
- this is still a problem in SB 7.x
- you can provide a consistent reproduction in 7.x (at https://storybook.new/ or per [repro docs](https://storybook.js.org/docs/react/contribute/how-to-reproduce)).
